test/Upgradibility.tests.ts

```diff
@@ -0,0 +1,267 @@
+    depositHash2 = dr2.hash;
+  }
+
+  it("Should be able to upgrade contracts", async function () {
+    (await upgrades.upgradeProxy(liquidityPool.address, await ethers.getContractFactory("LiquidityPool"))).deployed();
+    (
+      await upgrades.upgradeProxy(liquidityProviders.address, await ethers.getContractFactory("LiquidityProviders"))
+    ).deployed();
+    (
+      await upgrades.upgradeProxy(farmingContract.address, await ethers.getContractFactory("HyphenLiquidityFarming"))
+    ).deployed();
+    tokenManager = (await upgrades.deployProxy(await ethers.getContractFactory("TokenManager"), [
+      tf.address,
+      pauser.address,
+    ])) as TokenManager;
+    await tokenManager.deployed();
+
+    await liquidityPool.setTokenManager(tokenManager.address);
+    await liquidityProviders.setTokenManager(tokenManager.address);
+    await wlpm.setTokenManager(tokenManager.address);
+
+    // Add supported ERC20 token
+    await tokenManager
+      .connect(owner)
+      .addSupportedToken(tokenAddress, minTokenCap, maxTokenCap, equilibriumFee, maxFee, 0);
+
+    let tokenDepositConfig = {
+      min: minTokenCap,
+      max: maxTokenCap,
+    };
+    await tokenManager.connect(owner).setDepositConfig([1], [tokenAddress], [tokenDepositConfig]);
+
+    // Add supported Native token
+    await tokenManager
+      .connect(owner)
+      .addSupportedToken(NATIVE, minNativeTokenCap, maxNativeTokenCap, equilibriumFee, maxFee, 0);
+
+    let nativeDepositConfig = {
+      min: minTokenCap,
+      max: maxTokenCap,
+    };
+    await tokenManager.connect(owner).setDepositConfig([1], [NATIVE], [nativeDepositConfig]);
+
+    await tokenManager.changeExcessStateFee(NATIVE, excessStateFee);
+    await tokenManager.changeExcessStateFee(token.address, excessStateFee);
+  });
+
+  it("Should be able to unstake tokens", async function () {
+    await expect(farmingContract.withdraw(1, owner.address)).to.not.be.reverted;
+    await expect(farmingContract.withdraw(2, owner.address)).to.not.be.reverted;
+    expect(await lpToken.ownerOf(1)).to.equal(owner.address);
+    expect(await lpToken.ownerOf(2)).to.equal(owner.address);
+  });
+
+  it("Should be able to process exit transaction", async function () {
+    const tokenBalance = await token.balanceOf(owner.address);
+    const nativeBalance = await ethers.provider.getBalance(owner.address);
+
+    await expect(
+      liquidityPool.connect(executor).sendFundsToUser(token.address, parseEther("1"), owner.address, depositHash1, 0, 1)
+    ).to.not.be.reverted;
+
+    await expect(
+      liquidityPool.connect(executor).sendFundsToUser(NATIVE, parseEther("1"), owner.address, depositHash2, 0, 1)
+    ).to.not.be.reverted;
+
+    expect((await token.balanceOf(owner.address)).gte(tokenBalance)).to.be.true;
+    expect((await ethers.provider.getBalance(owner.address)).gte(nativeBalance)).to.be.true;
+  });
+
+  it("Should be able to withdraw liquidity", async function () {
```

Comment:
Add a test case to verify if token manager was changed after contracts upgrades.